src/components/PinInput/src/PinInput.vue

```diff
@@ -0,0 +1,308 @@
+			return `code-cell-${index}`;
+		},
+
+		// Focus on verification code cell input at given index.
+		focusOnCodeCell(index) {
+			const cellReference = this.$refs[this.getCodeCellRef(index)];
+			if (cellReference?.[0]?.focus) {
+				cellReference[0].focus();
+			}
+		},
+
+		onInputCode(event, index) {
+			if (!event.data) {
+				return;
+			}
+
+			// Only allow integers in input
+			const BASE_TEN = 10;
+			const inputValue = Number.isInteger(Number.parseInt(event.data, BASE_TEN)) ? event.data : '';
+
+			// One-time-code autofill is treated as an input, not a paste
+			if (this.attemptSplitCodeIntoInputs(inputValue, index)) {
+				return;
+			}
+
+			this.$set(this.code, index, inputValue);
+
+			const firstIncompleteCellIndex = this.findFirstIncompleteCodeCellIndex();
+
+			// eslint-disable-next-line no-magic-numbers
+			if (firstIncompleteCellIndex === -1) {
+				this.handleComplete(this.currentCode);
+				return;
+			}
+
+			// This allows us to auto jump to the next code cell input as the user types.
+			this.focusOnCodeCell(firstIncompleteCellIndex);
+		},
+
+		attemptSplitCodeIntoInputs(value, inputIndex) {
+			const BASE_TEN = 10;
+			if (inputIndex === 0
+				&& value?.length === this.pinLength
+				&& Number.isInteger(Number.parseInt(value, BASE_TEN))
+			) {
+				this.$set(this, 'code', value.split(''));
+
+				// Having a timeout here gives the user a chance to see their code before success/failure
+				const TIMEOUT_LENGTH_MS = 500;
+				setTimeout(() => { this.handleComplete(this.currentCode); }, TIMEOUT_LENGTH_MS);
+				return true;
+			}
+			return false;
+		},
+
+		handleComplete(code) {
+			this.$emit('complete', code);
+		},
+
+		onPasteCode(event, index) {
+			event.preventDefault();
+			const pastedValue = (event.clipboardData || window.clipboardData).getData('text');
+			this.attemptSplitCodeIntoInputs(pastedValue, index);
+		},
+
+		onDelete(event, index) {
+			// Reset current index, and then focus on previous cell
+			this.$set(this.code, index, '');
+
+			// eslint-disable-next-line no-magic-numbers
+			const newIndex = index - 1;
+			if (newIndex >= 0) {
+				this.focusOnCodeCell(newIndex);
+			}
+		},
+
+		onFocusCode(event, index) {
+			event.preventDefault();
+			this.$set(this.code, index, '');
+		},
+
+		resetcode() {
+			const newArray = (new Array(this.pinLength)).fill('');
+			this.$set(this, 'code', newArray);
+		},
+
+		shakeAndClearInputs() {
+			this.resetcode();
+			this.focusOnCodeCell(0);
+			this.isShaking = true;
+			const TIMEOUT_LENGTH_MS = 1000;
+			setTimeout(() => {
+				this.isShaking = false;
+			}, TIMEOUT_LENGTH_MS);
+		},
+	},
+};
+</script>
+
+<style module="$s">
+.PinInputContainer {
+	display: flex;
+	flex-wrap: nowrap;
+	align-items: center;
+	justify-content: space-between;
+	padding-bottom: 8px;
+}
+
+.CodeInput {
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	width: 46px;
+	height: 46px;
+	padding: 0;
+	font-weight: 500;
+	font-size: var(--font-step-0-size);
```

Comment:
this CSS variable is not defined in the `master` branch, it's only in the `beta` branch. Since Maker Theming is still in `beta` I wouldn't worry about this for now, you can set this to `font-size: inherit` if you'd like it to be controllable from the client OR to `font-size: 16px` which is sensible default for now.